src/libostree/ostree-repo-pull.c

```diff
@@ -2133,6 +2195,65 @@ repo_remote_fetch_summary (OstreeRepo    *self,
   return ret;
 }
 
+static void
+delta_superblock_fetch_on_complete (GObject        *object,
+                         GAsyncResult   *result,
+                         gpointer        user_data)
+{
+  OstreeFetcher *fetcher = (OstreeFetcher *)object;
+  FetchDeltaSuperBlockData *fetch_data = user_data;
+  OtPullData *pull_data = fetch_data->pull_data;
+  GError *local_error = NULL;
+  GError **error = &local_error;
+  GInputStream* input = _ostree_fetcher_stream_uri_finish (fetcher, result, error);
+  GBytes* delta_superblock_data = NULL;
+
+  if (input)
+    {
+      delta_superblock_data = g_input_stream_read_bytes ( input, OSTREE_MAX_METADATA_SIZE, pull_data->cancellable, error);
```

Comment:
That error is a pointer to local_error, which in turn is propagated at the end of the function,, like in all of the async callbacks. Is that not sufficient?

I was more worried about using ``pull_data->cancellable``, but apparently that's thread-safe.